### DISCUSSION TITLE
Silos and responsibilities of custom metrics

### DISCUSSION THREAD

**Discussion by xamroc** [argoproj/argo-rollouts#1595] [Q&A]
When I look at Argo Rollouts, I get the impression that it's a tool purely for developers. The most common examples I see from the community is using Prometheus.

However, given that ops and devs are separate teams with responsibilities like so:
- Dev: Write custom metrics
- Ops: Write logic for Argo Rollouts releases

From an ops perspective, they are heavily relying on dev teams to write good custom metrics for Argo Rollouts analysis. I think it would be better for ops teams to rely on `web` or `job` analysis on services to separate this dependency. But I'm open to other perspectives on how to better handle this.

*I'm aware that we should be more devops and break silos but sometimes this is not pragmatic. Either we do not have developers that know everything or not enough ops to help all dev teams

**Comment by kostis-codefresh** [comment]
I am not sure what is the question here.

First of all I wouldn't say that Argo Rollouts is for developers. On a big enterprise company someone could even expect developers to focus only on writing code and not bother themselves with Kubernetes controllers.

But regarding metrics, your application should already have metrics. An application should always have metrics for monitoring and maintenance. This is a completely unrelated to Argo Rollouts. The fact that Argo Rollouts takes advantage of existing metrics in order to promote a release is an added bonus.

If you try to adopt Argo Rollouts, and you realize that your applications don't have metrics  you can use, then it is best to focus on monitoring first before moving to Progressive delivery.
